### DISCUSSION TITLE
Jukebox Dashboard for Home Assistant

### DISCUSSION THREAD

**Discussion by OzGav** [music-assistant/support#2175] [Show and tell]
There have been calls for a jukebox mode for MA. While we wait to see if that is something that can be provided I though I would do a proof of concept. I have come up with this:
![image](https://github.com/music-assistant/support/assets/19848947/4a3502ff-7318-4d28-8411-5209f0f3cc68)

To use this you will need two custom components from HACS [ha-floorplan](https://experiencelovelace.github.io/ha-floorplan/) and [browser-mod](https://github.com/thomasloven/hass-browser_mod)

You can select the player by double clicking on a hidden button in that black rectangle at the centre bottom below the numbers. The player name will appear for 2 secs and then vanish. The available players must be hard coded into an input_select and those options must be identical to the player friendly names. You also need to know that I have the characters `ma_` in all of my MA entities so I can identify them easily and this uniqueness is used in templates to resolve the `entity_id` from the friendly name.

When clicking on the search area a custom pop up courtesy of browser-mod will appear for text entry

Once the eight fields are then populated you simply click on those fields to add that song to the queue.

I have put all of the parts that need to be created into a package file, the view is in a separate file and the SVG image is the third file in the ZIP.

[jukebox.zip](https://github.com/music-assistant/support/files/15157213/jukebox.zip)

I do all my work in YAML but you should be able to recreate all of this in the UI but you will need access to the HA file structure to upload the SVG.

**Comment by patienttruth** [comment]
That looks awesome. I'll definitely check this out! Great work!

**Comment by Topiatic** [comment]
Just saw this and need to add that your taste in music is supurb. A hair repeitive, but you got Brothers in Arms in there :).

**Comment by fsmythe** [comment]
Is it possible to refresh the links on the OP? Think they've broken after the adoption of MA into official maybe?

**Reply by OzGav** [reply]
done
